extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/output/RetryableS3OutputStream.java

```diff
@@ -269,52 +213,74 @@ public void close() throws IOException
     // Closeables are closed in LIFO order
     closer.register(() -> {
       // This should be emitted as a metric
+      long totalChunkSize = (currentChunk.id - 1) * chunkSize + currentChunk.length();
```

Comment:
Hmm, while this is correct, it seems better to simply add up the size of every chunk and not rely on the chunkSize here. If you use a map to maintain the futures, you could the total size of the successfully uploaded chunks.

Reply:
It allows us to not have a Map<Chunk, Future> unnecessarily and helps the rest of the code be cleaner and simpler, which I really liked. Can you please elaborate more on why you'd prefer adding them individually? Thanks!

PS: Can the chunkSize change in the middle of the operation? (in case of node failures or rolling upgrades where we update the chunk size or something along those lines?) 🤔

Reply:
> It allows us to not have a Map<Chunk, Future> unnecessarily

To me, having the map seems cleaner as we know which future belongs to which chunk. I don't see how it would affect the rest of the code as we just need to put stuff in the map and then iterate over it in `completeMultipartUpload()` and then while determining the size (which could even just be returned by the `completeMultipartUpload()` method).

That said, I don't really mind the current approach either. I guess the map is more of a personal preference in this case.

Reply:
> To me, having the map seems cleaner as we know which future belongs to which chunk

@kfaraz The returned future (if successful) contains UploadPartResult which contains the partNumber which is the chunk ID, if that helps.

Reply:
It's fine for now. We can worry about it when we really need to do something with the chunk. For now, let's just stick with the list.